core/api.cfc

```diff
@@ -1299,7 +1299,18 @@
 			<cfset func = metadata.functions[f] />
 			<!--- ignore hidden methods, if access is not set, assume public --->
 			<cfif not structKeyExists(func, "access") or (func.access neq "private" and func.access neq "package")>
-				<cfset arrayAppend(result.functions, func) />
+				<!--- check to see if this function is already in the list. If so, overwrite, otherwise append --->
+				<cfset foundFunc = False />
+				<cfloop from="1" to="#arrayLen( result.functions )#" index="g">
```

Comment:
This line made me realize that neither your `g` nor its predecessor `f` are properly var-scoped. While you're in here, do you mind adding some vars to the top of the function to properly scope these two variables?